src/weights.jl

```diff
@@ -291,3 +291,82 @@ end
 
 wmedian(v::RealVector, w::RealVector) = median(v, weights(w))
 wmedian{W<:Real}(v::RealVector, w::WeightVec{W}) = median(v, w)
+
+
+
+
+###### Weighted quantile #####
+# Variables m, g, low, high, correspond respectively to m, gamma, x[j], x[j+1] in the R help file for quantile
+
+function Base.quantile{W<:Real}(v::RealVector, p::RealVector, w::WeightVec{W})
+
+    isempty(v) && error("quantile of an empty array is undefined")
+    isempty(p) && throw(ArgumentError("empty quantile array"))
+
+    ppermute = sortperm(p)
+    p = p[ppermute]
+
+    # make sure the quantiles are in [0,1]
+    p = bound_quantiles(p)
+
+    if w.sum == 0
+        error("weight vector cannot sum to zero")
+    end
+    if length(v) != length(w)
+        error("data and weight vectors must be the same size. got $(length(v)) and $(length(w))")
+    end
+    for x in w.values
+        isnan(x) && error("weight vector cannot contain NaN entries")
+        x < 0 && error("weight vector cannot contain negative entries")
+    end
+    for x in v
+        isnan(x) && return x
+    end
+
+    lv = length(v)
+    lp = length(p)
+    m = 1 - p
+    index = (lv * p + m) * w.sum / lv
+    wt = w.values
+    vpermute = sortperm(v)
```

Comment:
I guess you could take inspiration from `sortrows`. Adding a version of it that would return the permutation has been requested: https://github.com/JuliaLang/julia/issues/9258

Unfortunately, it would require you to create a matrix holding a copy of `v` and `w`, which would be terribly wasteful AFAICT. So maybe you would need to write a custom function based on `sortperm` directly.